knowledge.py

```
@@ -0,0 +1,23 @@
+
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+DATA_DIR = os.path.join(BASE_DIR, "data")
+FILENAME = os.path.join(DATA_DIR, "knowledge.txt")
+
+def load_knowledge():
+    if os.path.exists(FILENAME):
+        with open(FILENAME, "r", encoding="utf-8") as f:
+            return json.load(f)
+    return {}
+
+def save_knowledge(knowledge):
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)